### ISSUE TITLE
Safari Bug: Microphone input not working

### ISSUE THREAD

**Issue by Amartya** [Amartya/TunePad#64]
Need to look more into this, but the voice input/microphone is not working. Might be a Dart compilation bug.

**Comment by Amartya** [comment]
This turned out to be really tricky. Dart as of now, uses an older call/method for getUserMedia that has been rendered obsolete by W3C (it will continue to work in Chrome/Firefox in the foreseeable future). Safari joined this getUserMedia party after W3C standardized the api call to navigator.mediadevices.getUserMedia (instead of navigator.getUserMedia). More details here: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/getUserMedia

Dart still compiles to the older api call. Till this is updated on the Dart end, not much we can do to fix this issue. I did try something, wherein, I called the JS function via Dart (using Dart's JS interop), and I was able to trigger the microphone on Safari. However, I was unable to handle the returned JSObject correctly :(. Casting it as a MediaStream fails. We can revisit this later, one option would be to offload the voxel's voice recording function to a Javascript fallback instead. 

For now, I have a modal show up that tells the user what is going on, and if he/she wishes to record audio, Firefox or Chrome is the best bet. 

This is part of pull request #86 

Dart's implementation of getUserMedia
<img width="758" alt="screen shot 2019-02-25 at 12 34 17 am" src="https://user-images.githubusercontent.com/279301/53322965-e6d7aa00-38aa-11e9-929f-a96a2fb32626.png">

My using Dart's JS interop to call the Javascript method with the new api call does trigger the microphone
<img width="791" alt="screen shot 2019-02-24 at 11 14 17 pm" src="https://user-images.githubusercontent.com/279301/53322983-f35c0280-38aa-11e9-9b2e-c4681ce84239.png">

Finally, a modal that just informs the user that the voice recording function is toast in Safari.
<img width="733" alt="screen shot 2019-02-25 at 2 52 50 am" src="https://user-images.githubusercontent.com/279301/53323022-0c64b380-38ab-11e9-8759-7f46bf97ce5b.png">
